app/models/miq_db_config.rb

```diff
@@ -175,69 +175,50 @@ def method_missing(m, *args)
     end
   end
 
-  def opt_file_for_conn_test(from_save, &blk)
-    raise "must pass block" unless block_given?
-    opts = self.class.raw_config['production']
+  def with_temporary_connection(config_hash = nil)
+    config_hash ||= self.class.raw_config['production']
 
-    begin
-      FileUtils.mkdir_p(File.dirname(DB_FILE))
-      File.delete(DB_FILE) if File.exist?(DB_FILE)
-      opts[:from_save] = from_save
-      File.open(DB_FILE, "wb") {|f|f.write(Base64.encode64(Marshal.dump(opts)))}
-      res = yield
-    ensure
-      File.delete(DB_FILE) if File.exist?(DB_FILE)
-    end
-    res
-  end
-
-  def delete_io_files
-    [IO_DOLLAR_STDOUT, IO_DOLLAR_STDERR].each {|f| File.delete(f) if File.exist?(f)}
+    before = ActiveRecord::Tasks::DatabaseTasks.database_configuration
+    ActiveRecord::Tasks::DatabaseTasks.database_configuration = config_hash
+    conn = ActiveRecord::Base.establish_connection(config_hash).connection
```

Comment:
@Fryguy you are correct!  I guess we'd need to call establish_connection again in the ensure (or possibly call `clear_active_connections`, but I'd have to double check)